LICENSE

```diff
@@ -1,502 +1,21 @@
-LIBRARY IS WITH YOU.  SHOULD THE LIBRARY PROVE DEFECTIVE, YOU ASSUME
-THE COST OF ALL NECESSARY SERVICING, REPAIR OR CORRECTION.
-
-  16. IN NO EVENT UNLESS REQUIRED BY APPLICABLE LAW OR AGREED TO IN
-WRITING WILL ANY COPYRIGHT HOLDER, OR ANY OTHER PARTY WHO MAY MODIFY
-AND/OR REDISTRIBUTE THE LIBRARY AS PERMITTED ABOVE, BE LIABLE TO YOU
-FOR DAMAGES, INCLUDING ANY GENERAL, SPECIAL, INCIDENTAL OR
-CONSEQUENTIAL DAMAGES ARISING OUT OF THE USE OR INABILITY TO USE THE
-LIBRARY (INCLUDING BUT NOT LIMITED TO LOSS OF DATA OR DATA BEING
-RENDERED INACCURATE OR LOSSES SUSTAINED BY YOU OR THIRD PARTIES OR A
-FAILURE OF THE LIBRARY TO OPERATE WITH ANY OTHER SOFTWARE), EVEN IF
-SUCH HOLDER OR OTHER PARTY HAS BEEN ADVISED OF THE POSSIBILITY OF SUCH
-DAMAGES.
-
-                     END OF TERMS AND CONDITIONS
-
-           How to Apply These Terms to Your New Libraries
-
-  If you develop a new library, and you want it to be of the greatest
-possible use to the public, we recommend making it free software that
-everyone can redistribute and change.  You can do so by permitting
-redistribution under these terms (or, alternatively, under the terms of the
-ordinary General Public License).
-
-  To apply these terms, attach the following notices to the library.  It is
-safest to attach them to the start of each source file to most effectively
-convey the exclusion of warranty; and each file should have at least the
-"copyright" line and a pointer to where the full notice is found.
-
-    <one line to give the library's name and a brief idea of what it does.>
-    Copyright (C) <year>  <name of author>
-
-    This library is free software; you can redistribute it and/or
-    modify it under the terms of the GNU Lesser General Public
-    License as published by the Free Software Foundation; either
-    version 2.1 of the License, or (at your option) any later version.
-
-    This library is distributed in the hope that it will be useful,
-    but WITHOUT ANY WARRANTY; without even the implied warranty of
-    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
-    Lesser General Public License for more details.
-
-    You should have received a copy of the GNU Lesser General Public
-    License along with this library; if not, write to the Free Software
-    Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
-
-Also add information on how to contact you by electronic and paper mail.
-
-You should also get your employer (if you work as a programmer) or your
-school, if any, to sign a "copyright disclaimer" for the library, if
-necessary.  Here is a sample; alter the names:
-
-  Yoyodyne, Inc., hereby disclaims all copyright interest in the
-  library `Frob' (a library for tweaking knobs) written by James Random Hacker.
-
-  <signature of Ty Coon>, 1 April 1990
-  Ty Coon, President of Vice
-
-That's all there is to it!
+MIT License
+
+Copyright (c) 2020 Pascal Merz, Michael R. Shirts
```

Comment:
I have no idea

1. Which years should be covered. LGPL doesn't seem to include a copyright (?) but it was definitely worked on since 2017.
2. Whose names should be on it

I skimmed `*/LICENSE` in the folder I throw my clones into and ended up only more confused. A combination of authors, PIs, labs, unversities, projects, etc. are listed on various MIT licenses I use. I am certainly not a lawyer and don't know the formally correct thing to do here

Reply:
Maybe just the `Shirts Group` would suffice? Probably would be best for @mrshirts to decide.

Reply:
License and copyright are independent; you always have copyright (under US code at least) and a license is how you give permission to copy it (otherwise, they only have fair use, which doesn't include whole working code). 

My understanding is that the copyright officially resides with the University of Colorado Boulder because of the employee-employer arrangement - but they allow us to license code we write as permissively as we want to if we judge it is in the best interest. 

Year - that's a good question.  Looking at the gitlog history, the repo looks like it evolved from the check-ensemble svn repository! If there is remaining code from check_ensemble, we may have to add a "Portions Copyright the University of Virginia", starting in 2011.  If it's (essentially) fresh now, we should just do University of Colorado Boulder, and have the years 2017-2020.

Reply:
Here are the authors, according to grepping the git history (my only commit is in this branch)

```
Matthew W. Thompson
Michael Shirts
Pascal Merz
Pascal T. Merz
SimonBoothroyd
mrshirts
```

> Maybe just the Shirts Group would suffice?

I'm inclined to agree here, I think?

> If it's (essentially) fresh now

This has been an interesting git history to poke around in; ada32fd61ccdc4a96fa1f215e1a4d3df846502c2 is the first commit at CU and would probably be the starting point for answering this question?

Reply:
> I'm inclined to agree here, I think?

Copyright should be "Copyright (c) 2017-2020 University of Colorado Boulder".  Unless @ptmerz thinks that some of the code remains from checkensemble.

Authors can then be listed as the people involved: which could be all the members of the team that end up working on it. "Shirts Group" isn't aa great collective identifier since the group membership changes over time.

Reply:
Ok, so I updated the copyright to

```
Copyright (c) 2017-2021 University of Colorado Boulder
          (c) 2012      The University of Virginia
```

There is still some low-level implementations of the ensemble checking which are essentially slightly modernized versions of what you wrote in the checkensemble code. So we probably need to keep UVa in. Also, 2012 is what the files in checkensemble were originally copyrighted as - let me know if you think that should span more years @mrshirts.